app/models/service_request.rb

```diff
@@ -598,6 +598,16 @@ def has_non_first_draft_ssrs?
     sub_service_requests.where.not(status: 'first_draft').any?
   end
 
+  def ssrs_associated_with_service_provider (service_provider)
```

Comment:
I think if you used #map here, you could condense this method a bit. Map will create an array and return it. So you wouldn't need to instantiate an empty array.